platform/posix/utest/reconnect_utest.c

```diff
@@ -45,19 +48,23 @@ int suiteTearDown( int numFailures )
 static void verifyReconnectParamsAfterReset( void )
 {
     TEST_ASSERT_EQUAL( 0, reconnectParams.attemptsDone );
-    TEST_ASSERT_GREATER_OR_EQUAL_INT32( INITIAL_RECONNECT_BACKOFF_SECONDS,
-                                        reconnectParams.nextJitterMax );
-    TEST_ASSERT_LESS_THAN_INT32( ( INITIAL_RECONNECT_BACKOFF_SECONDS +
-                                   MAX_JITTER_VALUE_SECONDS ),
-                                 reconnectParams.nextJitterMax );
+    TEST_ASSERT_EQUAL( JITTER, reconnectParams.nextJitterMax - JITTER );
+    TEST_ASSERT_GREATER_OR_EQUAL_UINT32( INITIAL_RECONNECT_BACKOFF_SECONDS,
+                                         reconnectParams.nextJitterMax );
+    TEST_ASSERT_LESS_THAN_UINT32( ( INITIAL_RECONNECT_BACKOFF_SECONDS +
+                                    MAX_JITTER_VALUE_SECONDS ),
+                                  reconnectParams.nextJitterMax );
+    TEST_ASSERT_LESS_THAN_UINT32( MAX_RECONNECT_ATTEMPTS,
+                                  reconnectParams.attemptsDone );
 }
 
 /**
  * @brief Test that #Transport_ReconnectParamsReset initializes the #nextJitterMax
- * to be equal to the some configurable backoff plus jitter.
+ * to be equal to some configurable backoff plus jitter.
  */
 void test_Transport_ReconnectParamsReset_Sets_Jitter_Correctly( void )
 {
+    rand_ExpectAndReturn( JITTER );
```

Comment:
`rand` is not required by `Transport_ReconnectParamsReset`

Reply:
Yes it is...

Reply:
Oops my bad. In that case, there can be a global mock configuration for the `rand` in `setUp`

Reply:
obfuscation